### ISSUE TITLE
Move editor/ and engine/ to a common src/ directory

### ISSUE THREAD

**Issue by craftablescience** [craftablescience/ChiraEngine#36]
### Component to Improve

Other

### Describe Your Suggestion

The editor and engine code should be moved to a common src directory. This refactor is impossible to do before PRs and branches are merged into main.

**Comment by ashifolfi** [comment]
@craftablescience I think it's actually better to keep the main engine and the editor portions separate for a code organization purpose. having all of the editor code mingled in the main engine code would be extremely messy and would make removing the editor components difficult for anyone who wants to remove them from their game.

Not to mention it's just confusing.

**Comment by craftablescience** [comment]
I'm not saying to merge the code, I'm saying to move the source code to a `src/` folder like how every other C++ project is organized

**Comment by ashifolfi** [comment]
ah

**Comment by craftablescience** [comment]
All existing PRs would need to be merged first so its not a big priority

**Comment by craftablescience** [comment]
I think this isn't too necessary now, it would just be another folder to open

Godot doesn't have a src folder and they're doing pretty well organization wise
